### PR TITLE
feat: support tiflash backup and restore during volume snapshot

### LINKED ISSUE #4807: TiKV pods can't be created when execute EBS snapshot restore with Tiflash
## Bug Report

**What version of Kubernetes are you using?**

**What version of TiDB Operator are you using?**

v1.4.0-beta.4

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

tiflash and pd are running, other pods are not created.

**What did you do?**

1. create a tc with 1 pd 1 tikv 1 tidb 1 tiflash.
2. create an EBS snapshot restore.

**What did you expect to see?**

restore cluster successfully.

**What did you see instead?**

pods of tiflash and pd are created, but pods of tikv and tidb are not created. There is a error log in TiDB Operator: W1221 03:11:18.903080       1 tiflash_member_manager.go:699] Fail to GetStores for TidbCluster tidb-cluster/basic: Error response 500 URL http://basic-pd.tidb-cluster:2379/pd/api/v1/stores,body response: "[PD:cluster:ErrNotBootstrapped]TiKV cluster not bootstrapped, please start TiKV first"

### PR DESCRIPTION
**Pull request by fengou1**
### What problem does this PR solve?
add tiflash support for volume-snapshot backup and restore

Closes #4807 

### What is changed and how does it work?
1. after BR 1st restore done, start TiKV
2. BR 2nd restore will set pd allocate id and after that, the 2nd restore unmask the pd recovering mark.
3. TiFlash detects if PD recovering mark is unset, start TiFlash
4. wait BR 2nd restore done

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- backup without tiflash - PASS
- restore without tiflash - PASS
- backup without tiflash - PASS
- restore with tiflash - PASS
- backup with tiflash - PASS
- restore with tiflash - PASS
- backup with tiflash - PASS
- restore without tif lash - PASS
- 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/backup/restore/restore_manager.go [hunk 1]
```diff
@@ -236,8 +247,70 @@ func (rm *restoreManager) readRestoreMetaFromExternalStorage(r *v1alpha1.Restore
...
+		return 0, "FileNotExists", fmt.Errorf("%s does not exist", constants.MetaFile)
+	}
+
+	restoreMeta, err := externalStorage.ReadAll(ctx, constants.MetaFile)
+	if err != nil {
+		return 0, "ReadAllOnExternalStorageFailed", err
+	}
+
+	metaInfo := &backuputil.EBSBasedBRMeta{}
+	err = json.Unmarshal(restoreMeta, metaInfo)
```
Comment (WangLe1321) [inline pkg/backup/restore/restore_manager.go:301]: Do we have a common method to get restore meta from external storage? If not exist, I suggest to split it to a common method.

File: pkg/pdapi/fake_pdapi.go [hunk 2]
```diff
@@ -270,3 +271,13 @@ func (c *FakePDClient) GetAutoscalingPlans(strategy Strategy) ([]Plan, error) {
 	}
 	return nil, nil
 }
+
+func (c *FakePDClient) GetRecoveringMark() (bool, error) {
+	action := &Action{}
+	result, err := c.fakeAPI(GetRecoveringMarkActionType, action)
+	if err != nil {
+		return result.(bool), err
```
Comment (WangLe1321) [inline pkg/pdapi/fake_pdapi.go:279]: I suggest to return `false, err`, I am not sure whether the `result` is `nil` if `err != nil`.

Comment (fengou1) [reply pkg/pdapi/fake_pdapi.go]: fix in https://github.com/pingcap/tidb-operator/pull/4812/commits/30ac10ba55a7b72b5780652c5163c320d99a4453

File: pkg/backup/restore/restore_manager.go (hunk 1 above)
Comment (fengou1) [reply pkg/backup/restore/restore_manager.go]: fixed in https://github.com/pingcap/tidb-operator/pull/4812/commits/30ac10ba55a7b72b5780652c5163c320d99a4453

**Review by WangLe1321 — APPROVED** [review]
LGTM

**Review by WizardXiao — APPROVED** [review]

**Pull request merged**